services/pull/pull.go

```diff
@@ -516,6 +517,8 @@ func CloseRepoBranchesPulls(doer *models.User, repo *models.Repository) error {
 	return nil
 }
 
+var commitMessageTrailersPattern = regexp.MustCompile(`(?:^|\n\n)(?:[\w-]+: [^\n]+\n*)+$`)
```

Comment:
OK testing from git interpret-trailers I think this needs to be, at least:

```suggestion
var commitMessageTrailersPattern = regexp.MustCompile(`(?:^|\n\n)(?:[\w-]+[:=][^\n]+\n)*(?:[\w-]+: [^\n]+)\n*$`)
```

How trailers can be folded, see https://git-scm.com/docs/git-interpret-trailers

> When reading trailers, there can be whitespaces after the token, the separator and the value. There can also be whitespaces inside the token and the value. The value may be split over multiple lines with each subsequent line starting with whitespace, like the "folding" in RFC 822.

Which would make this:

```suggestion
var commitMessageTrailersPattern = regexp.MustCompile(`(?:^|\n\n)(?:[\w-]+[:=][^\n]+\n(?: [^\n]*\n)*)*(?:[\w-]+: [^\n]+(?:\n [^\n]*)*)\n*$`)
```

(I think)

Reply:
```
echo -n 'Fix #1546                                                                         [19:26:27]

Signed-off-by: Not a trailer

Signed-off-by:Alice <alice@example.com>
Signed-off-by:   Bob <bob@example.com>
' | git interpret-trailers --parse
```

```
echo -n 'Fix #1546                                                                         [19:26:27]

Signed-off-by:Not a trailer

Signed-off-by:Alice <alice@example.com>
Signed-off-by:   Bob <bob@example.com>' | git interpret-trailers --parse
```

```
echo -n 'Fix #1546

Signed-off-by:Not a trailer

Signed-off-by:Alice <alice@example.com>
Signed-off-by:   Bob <bob@example.com>


' | git interpret-trailers --parse
```

```
echo -n 'Fix #1546

Signed-off-by:Not a trailer

Signed-off-by:Alice <alice@example.com>
Folded: This is a folded value
 with folded results
   preceding spaces?
Signed-off-by:   Bob <bob@example.com>' | git interpret-trailers --parse
```